core/Azure.Mcp.Core/tests/Azure.Mcp.Core.UnitTests/Client/MockClientTests.cs

```diff
@@ -180,6 +180,45 @@ await Invoke_Request_To_Server(
             });
     }
 
+    [Fact]
+    public async Task Invoke_Invalid_Tool_Returns_Error()
+    {
+        await Invoke_Request_To_Server(
+            method: "tools/call",
+            new ServerCapabilities
```

Comment:
FYI @stephentoub, this PR will conflict with the changes being made in #576 (unless this API continues to exist and your PR is updating to more optimal designs)